api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java

```diff
@@ -136,20 +211,49 @@ public DiagnosticReport saveFHIRDiagnosticReport(DiagnosticReport diagnosticRepo
 		if (!codingList.isEmpty()) {
 			encounterType = codingList.get(0).getCode();
 		}
-		omrsDiagnosticReport.setEncounterType(FHIRUtils.getEncounterType(encounterType));
+		omrsDiagnosticReport.setEncounterType(getEncounterType(encounterType));
 
 		// Set `Diagnosis[x]->DateTime` as valueDateTime in an Obs
 		// Set `Diagnosis[x]->Period` as valueDateTime in an Obs
 
+		// Create resource in OpenMRS Database
+		Encounter omrsEncounter = encounterService.saveEncounter(omrsDiagnosticReport);
+
 		// Set parsed obsSet (`Result` as Set of Obs)
 		// Set Binary Obs Handler which used to store `PresentedForm`
+		for (AttachmentDt attachment : diagnosticReport.getPresentedForm()) {
+			int conceptId = FHIRUtils.getDiagnosticReportPresentedFormConcept().getConceptId();
+			if (attachment.getCreation() == null) {
+				attachment.setCreation(diagnosticReport.getIssuedElement());
+			}
+			Obs complexObs = saveComplexData(omrsDiagnosticReport, conceptId, omrsPatient, attachment);
+			obsList.add(complexObs);
+		}
+		//TODO: Is it necessary?  omrsDiagnosticReport.setObs(obsList);
```

Comment:
@milankarunarathne are you settting the encounter reference when you saving the obs
?
You have to create a encounter. If encounter created later then you need to update obs again